var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -46,68 +51,112 @@ class NetcdfC(CMakePackage, AutotoolsPackage):
     variant("dap", default=False, description="Enable DAP support")
+    variant("byterange", default=False, description="Enable byte-range I/O")
     variant("jna", default=False, description="Enable JNA support")
     variant("fsync", default=False, description="Enable fsync support")
-    variant("zstd", default=True, description="Enable ZStandard compression", when="@4.9.0:")
+    variant("nczarr_zip", default=False, description="Enable NCZarr zipfile format storage")
     variant("optimize", default=True, description="Enable -O2 for a more optimized lib")
 
-    # It's unclear if cdmremote can be enabled if '--enable-netcdf-4' is passed
-    # to the configure script. Since netcdf-4 support is mandatory we comment
-    # this variant out.
-    # variant('cdmremote', default=False,
-    #         description='Enable CDM Remote support')
+    variant("szip", default=True, description="Enable Szip compression plugin")
+    variant("blosc", default=True, description="Enable Blosc compression plugin")
+    variant("zstd", default=True, description="Enable Zstandard compression plugin")
+
+    with when("build_system=cmake"):
+        # Based on the versions required by the root CMakeLists.txt:
+        depends_on("cmake@2.8.12:", type="build", when="@4.3.3:4.3")
+        depends_on("cmake@2.8.11:", type="build", when="@4.4.0:")
+        depends_on("cmake@3.6.1:", type="build", when="@4.5.0:")
+        depends_on("cmake@3.12:", type="build", when="@4.9.0:")
+        # Starting version 4.9.1, nczarr_test/CMakeLists.txt relies on the FILE_PERMISSIONS feature
+        # of the configure_file command, which is only available starting CMake 3.20:
+        depends_on("cmake@3.20:", type="test", when="@4.9.1:")
 
-    # The patch for 4.7.0 touches configure.ac. See force_autoreconf below.
     with when("build_system=autotools"):
-        depends_on("autoconf", type="build", when="@4.7.0,main")
-        depends_on("automake", type="build", when="@4.7.0,main")
-        depends_on("libtool", type="build", when="@4.7.0,main")
-    # CMake system can use m4, but Windows does not yet support
-    depends_on("m4", type="build", when=sys.platform != "win32")
+        for __s in itertools.chain(["@main"], _force_autoreconf_when):
+            with when(__s):
+                depends_on("autoconf", type="build")
+                depends_on("automake", type="build")
+                depends_on("libtool", type="build")
+                depends_on("m4", type="build")
+        del __s
+
+    # M4 is also needed for the source and man file generation. All the generated source files are
+    # included in the release tarballs starting at least the oldest supported version:
+    depends_on("m4", type="build", when="@main")
+
+    # The man files are included in the release tarballs starting version 4.5.0 but they are not
+    # needed for the Windows platform:
+    for __p in ["darwin", "cray", "linux"]:
```

Comment:
Why the double underscore? I need to review if that obscure name mangling Python feature kicks in here...

Reply:
The first time I tried to create a loop in the package scope didn't go well (see [here](https://github.com/spack/spack/blob/32808f4fb51196634a6529e75ce2d81a7e3de229/var/spack/repos/builtin/packages/knem/package.py#L53-L57)). It took me a while to understand that I should not use `for compiler in` there. Since then, I try to name variables in this scope so that they have minimal chance to collide with something else. Later, I realised that the variable remain with the class instances and started explicitly removing them.

Reply:
@alalazo the name mangling this would effect this variable, but only outside of this scope, so while this does eschew PEP conventions it should be fine considering we don't need to access `__p` outside of this scope. That being said, @skosukhin is there some reason something like `plat` or `spack_host_platform` or something more within python naming conventions wouldn't work?

For reference/convenience: https://peps.python.org/pep-0008/#method-names-and-instance-variables

Reply:
@skosukhin Sure, I agree with that. Deleting is zealous, but not wrong. I was just wondering why the double underscore instead of a single one :slightly_smiling_face: